### PR TITLE
Remove deprecated SchemaPath

### PR DESCRIPTION
**Pull request by PeterSuna**

### CODE REVIEW

File: src/main/java/io/lighty/yang/validator/formats/JsonTree.java [hunk 1]
```diff
@@ -252,36 +281,49 @@ public Help getHelp() {
         return Optional.empty();
     }
 
-    private JSONObject resolveChildMetadata(final DataSchemaNode node, final Optional<Boolean> isConfig) {
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/JsonTree.java:255]: We can probably invoke stack.enter and stack.exit in the beginning and in the end of this method.
That way we can eliminate need to call enter and exit methods before and after this one.

```
stack.enter()
resolveChildMetadata()
stack.exit
```

becomes to be:

```
resolveChildMetadata()
```

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 2]
```diff
@@ -162,46 +175,55 @@ private void printLines(final List<Line> lines) {
...
             lines.add(htmlLine);
             resolveChildNodes(lines, new ArrayList<>(ids), node, RpcInputOutput.OTHER, removeChoiceQnames,
-                    Collections.emptyList());
+                    Collections.emptyList(), schemaInferenceStack);
+            schemaInferenceStack.exit();
         }
         return lines;
     }
 
     private List<Line> getAugmentationNodesLines(final ArrayList<Integer> ids, final AugmentationSchemaNode augNode,
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/JsTree.java:186]: You can change final ArrayList<Integer> ids to final List<Integer> ids here, too.

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: src/main/java/io/lighty/yang/validator/simplify/SchemaSelector.java [hunk 3]
```diff
@@ -67,44 +68,70 @@ private void fillUsedSchema(final InputStream input, final SchemaTree st)
...
+                    schemaInferenceStack.toSchemaNodeIdentifier());
             final Collection<? extends CaseSchemaNode> cases = ((ChoiceSchemaNode) node).getCases();
             for (DataSchemaNode singelCase : cases) {
-                resolveChildNodes(childSchemaTree, singelCase, false, false);
+                resolveChildNodes(childSchemaTree, singelCase, false, false, schemaInferenceStack);
             }
+            schemaInferenceStack.exit();
+        }
+
+        if (node instanceof DataTreeEffectiveStatement) {
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/simplify/SchemaSelector.java:109]: Why this new if is added?

`if (node instanceof DataTreeEffectiveStatement)`

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: src/main/java/io/lighty/yang/validator/formats/JsonTree.java (hunk 1 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsonTree.java:255]: Done

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java (hunk 2 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java]: Done

File: src/main/java/io/lighty/yang/validator/simplify/SchemaSelector.java (hunk 3 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/simplify/SchemaSelector.java]: I made this little complicated, I have reworked this. This IF create SchemaTree for elements which were not matched in other If conditions. Before SchemaTree was created on beginning of method. I was thinking that for ChoiceSchemaNode I have to use schemaInferenceStack.enterChoice(node.getQName()), but this is not true.

File: src/main/java/io/lighty/yang/validator/simplify/SchemaTree.java [hunk 4]
```diff
@@ -7,35 +7,41 @@
...
-    private final QName qname;
+
+    public static final Absolute ROOT = Absolute.of(QName.create("root", "root"));
+
+    private final Absolute absolutePath;
     private final DataSchemaNode schemaNode;
     private final boolean isRootNode;
     private final boolean isAugmenting;
     private final ActionDefinition actionNode;
+    private final LinkedHashSet<SchemaTree> children = new LinkedHashSet<>();
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/simplify/SchemaTree.java]: SchemaTree can be identified with absolutePath attribute. This attribute is used in equals() and hash() method, so it is behaving in a same way and make it easier to use.

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java [hunk 5]
```diff
@@ -780,17 +759,13 @@ private String buildTypeDefinitionPath(final TypeDefinition<?> typeDefinition,
             final SchemaInferenceStack schemaIS) {
         if (schemaIS.toInference().statementPath().size() == 1
                 && schemaIS.toInference().statementPath().get(0) instanceof TypedefEffectiveStatement) {
-            return schemaIS.toSchemaPath().toString();
+            return schemaIS.toString();
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:762]: Sure? What is a replacement for SchemaPath?

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 6]
```diff
@@ -61,23 +60,23 @@ public void emitFormat() {
...
-            for (AugmentationSchemaNode augNode : module.getAugmentations()) {
+            for (final AugmentationSchemaNode augNode : module.getAugmentations()) {
                 printLines(getAugmentationNodesLines(singletonListInitializer.getSingletonListWithIncreasedValue(),
-                        augNode, removeChoiceQnames));
+                        augNode, schemaInferenceStack));
             }
 
             // Rpcs
-            printLines(getRpcsLines(singletonListInitializer, module, removeChoiceQnames));
+            printLines(getRpcsLines(singletonListInitializer, module, schemaInferenceStack));
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/JsTree.java:76]: RPCs have absolute patch consisting just of its QName. We do not need stack.

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 7]
```diff
@@ -61,23 +60,23 @@ public void emitFormat() {
...
 
             // Notifications
-            printLines(getNotificationsLines(singletonListInitializer, module, removeChoiceQnames));
+            printLines(getNotificationsLines(singletonListInitializer, module, schemaInferenceStack));
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/JsTree.java:79]: Notifications have absolute patch consisting just of its QName. We do not need stack.

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 8]
```diff
@@ -61,23 +60,23 @@ public void emitFormat() {
...
-            SingletonListInitializer singletonListInitializer = new SingletonListInitializer(1);
+            final SingletonListInitializer singletonListInitializer = new SingletonListInitializer(1);
 
+            final SchemaInferenceStack schemaInferenceStack = SchemaInferenceStack.of(this.schemaContext);
             // Nodes
-            printLines(getChildNodesLines(singletonListInitializer, module, removeChoiceQnames));
+            printLines(getChildNodesLines(singletonListInitializer, module, schemaInferenceStack));
 
             // Augmentations
-            for (AugmentationSchemaNode augNode : module.getAugmentations()) {
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/JsTree.java:71]: What is the absolute path of augmentation?

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 9]
```diff
@@ -61,23 +60,23 @@ public void emitFormat() {
...
             final Module module = this.schemaContext.findModule(source.getName(), source.getRevision())
                     .orElseThrow(() -> new NotFoundException("Module", source.getName()));
-            final List<Integer> removeChoiceQnames = new ArrayList<>();
-            SingletonListInitializer singletonListInitializer = new SingletonListInitializer(1);
+            final SingletonListInitializer singletonListInitializer = new SingletonListInitializer(1);
 
+            final SchemaInferenceStack schemaInferenceStack = SchemaInferenceStack.of(this.schemaContext);
             // Nodes
-            printLines(getChildNodesLines(singletonListInitializer, module, removeChoiceQnames));
+            printLines(getChildNodesLines(singletonListInitializer, module, schemaInferenceStack));
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/JsTree.java:67]: Direct children of module - do we need stack?

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 10]
```diff
@@ -90,46 +89,52 @@ public void emitFormat() {
...
-            final Module module, final List<Integer> removeChoiceQnames) {
-        List<Line> lines = new ArrayList<>();
-        for (NotificationDefinition node : module.getNotifications()) {
-            final ArrayList<Integer> ids = singletonListInitializer.getSingletonListWithIncreasedValue();
-            LyvNodeData lyvNodeData = new LyvNodeData(this.schemaContext, node, Collections.emptyList());
-            HtmlLine htmlLine = new HtmlLine(new ArrayList<>(ids), lyvNodeData, RpcInputOutput.OTHER,
-                    removeChoiceQnames, namespacePrefix);
+            final Module module, final SchemaInferenceStack schemaIS) {
+        final List<Line> lines = new ArrayList<>();
+        for (final NotificationDefinition node : module.getNotifications()) {
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/JsTree.java:100]: Why to enter stack - we have full path from module.

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 11]
```diff
@@ -90,46 +89,52 @@ public void emitFormat() {
...
-            final Module module, final List<Integer> removeChoiceQnames) {
-        List<Line> lines = new ArrayList<>();
-        for (RpcDefinition node : module.getRpcs()) {
-            final ArrayList<Integer> rpcId = singletonListInitializer.getSingletonListWithIncreasedValue();
-            LyvNodeData lyvNodeData = new LyvNodeData(this.schemaContext, node, Collections.emptyList());
-            HtmlLine htmlLine = new HtmlLine(rpcId, lyvNodeData, RpcInputOutput.OTHER, removeChoiceQnames,
-                    namespacePrefix);
+            final Module module, final SchemaInferenceStack schemaIS) {
+        final List<Line> lines = new ArrayList<>();
+        for (final RpcDefinition node : module.getRpcs()) {
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/JsTree.java:118]: We have full path from module.

File: src/main/java/io/lighty/yang/validator/formats/Line.java [hunk 12]
```diff
@@ -45,70 +43,48 @@
...
+        resolveFlag(node, lyvNodeData.getAbsolutePath(), lyvNodeData.getContext());
         resolvePathAndType(node);
         resolveKeys(node);
         resolveIfFeatures(node);
     }
 
-    protected abstract void resolveFlag(SchemaNode node, SchemaContext context);
+    protected abstract void resolveFlag(SchemaNode node, Absolute absolutePath, EffectiveModelContext context);
 
-    protected void resolveFlagForDataSchemaNode(final SchemaNode node, final SchemaContext context,
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/Line.java:83]: We do not solve isConfig problem this in this PR.

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java [hunk 13]
```diff
@@ -149,48 +148,48 @@ private void printSchema(final SchemaTree tree) {
     private void printGroupings(final HashMap<GroupingDefinition, Set<SchemaTree>> trees) {
         for (final Map.Entry<GroupingDefinition, Set<SchemaTree>> tree : trees.entrySet()) {
             printer.openStatement(Statement.GROUPING, tree.getKey().getQName().getLocalName());
-            for (SchemaTree entry : tree.getValue()) {
-                doPrintSchema(true, entry, tree.getKey().getPath().getLastComponent().getLocalName(),
-                        new HashMap<>());
+            for (final SchemaTree entry : tree.getValue()) {
+                doPrintSchema(true, entry, tree.getKey().getQName().getLocalName(), new HashMap<>());
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java:152]: I am not sure if getQName() is repleacement for getPath().

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java [hunk 14]
```diff
@@ -301,7 +296,7 @@ private void doPrintSchema(final boolean isPrintingAllowed, final SchemaTree tre
         if (isPrintingAllowed) {
             if (schemaNode instanceof ContainerLike) {
                 printer.openStatement(Statement.CONTAINER, schemaNode.getQName().getLocalName());
-                printer.printConfig(schemaNode.isConfiguration());
+                printer.printConfig(schemaNode.effectiveConfig().orElse(Boolean.TRUE));
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java:299]: No, we do not change isConfiguration().

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java [hunk 15]
```diff
@@ -454,21 +448,15 @@ private void printImports() {
         }
     }
 
-    private static boolean isSchemaNodePathEqualsToDataSchemaNodePath(final DataSchemaNode schemaNode,
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java:457]: Why are those removed?

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/StatementPrinter.java [hunk 16]
```diff
@@ -21,11 +21,6 @@ void openStatement(final Statement type, final String name) {
         indentation++;
     }
 
-    void openStatement(final String name) {
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/yang/printer/StatementPrinter.java:24]: Why removed?

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/Util.java [hunk 17]
```diff
@@ -22,9 +22,5 @@ private Util() {
         }
         return typeDef;
     }
-
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/yang/printer/Util.java:25]: Why removed?

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: src/main/java/io/lighty/yang/validator/formats/JsonTree.java [hunk 18]
```diff
@@ -118,88 +120,115 @@ public void emitFormat() {
...
                     }
+                    schemaInferenceStack.exit();
                 }
 
-                appendActionsToAugmentationJson(augmentation, augmentationJson);
-                appendNotificationsToAugmentationJson(module, augmentationJson);
+                appendActionsToAugmentationJson(augmentation, augmentationJson, schemaInferenceStack);
+                appendNotificationsToAugmentationJson(module, augmentationJson, schemaInferenceStack);
                 jsonTree.append(AUGMENTS, augmentationJson);
+                schemaInferenceStack.exit();
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/JsonTree.java:135]: We do not need this line - we are entering augmentation target with full path.

File: src/main/java/io/lighty/yang/validator/formats/JsonTree.java [hunk 19]
```diff
@@ -252,36 +281,49 @@ public Help getHelp() {
         return Optional.empty();
     }
 
-    private JSONObject resolveChildMetadata(final DataSchemaNode node, final Optional<Boolean> isConfig) {
+    private JSONObject resolveChildMetadata(final DataSchemaNode node, final Optional<Boolean> isConfig,
+            SchemaInferenceStack schemaInferenceStack) {
         final JSONObject jsonModuleChild = new JSONObject();
         jsonModuleChild.put(NAME, node.getQName().getLocalName());
-        jsonModuleChild.put(CONFIG, isConfig.orElse(node.isConfiguration()));
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/JsonTree.java:258]: Revert this line.

File: src/main/java/io/lighty/yang/validator/formats/JsonTree.java [hunk 20]
```diff
@@ -348,8 +390,8 @@ private JSONObject resolveModuleMetadata(final Module module) {
         return jsonModuleMetadata;
     }
 
-    private String resolvePath(final SchemaPath schemaPath) {
-        final Iterable<QName> pathFromRoot = schemaPath.getPathFromRoot();
+    private String resolvePath(final Absolute absolutePath) {
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/JsonTree.java:377]: We can delete this method because Absolute extends SchemaNodeIdentifier.

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: src/main/java/io/lighty/yang/validator/simplify/stream/TrackingXmlParserStream.java [hunk 21]
```diff
@@ -135,8 +134,8 @@ public TrackingXmlParserStream parse(final XMLStreamReader reader) throws XMLStr
             } else {
                 throw new IllegalStateException("Unsupported schema node type " + parentNode.getClass() + ".");
             }
-
-            read(reader, nodeDataWithSchema, reader.getLocalName(), tree);
+            final SchemaInferenceStack schemaIS = SchemaInferenceStack.of(codecs.getEffectiveModelContext());
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/simplify/stream/TrackingXmlParserStream.java:138]: We do not need a stack - this class always begins with schemaroot - you can change parentNode type in constructor to effectivemodelcontext.

File: src/main/java/io/lighty/yang/validator/formats/ConsoleLine.java [hunk 22]
```diff
@@ -27,24 +27,24 @@
     private final List<Boolean> isConnected;
 
     ConsoleLine(final List<Boolean> isConnected, final LyvNodeData lyvNodeData, RpcInputOutput inputOutput,
-            final List<QName> removeChoiceQname, final Map<XMLNamespace, String> namespacePrefix) {
-        super(lyvNodeData, inputOutput, removeChoiceQname, namespacePrefix);
+            final Map<XMLNamespace, String> namespacePrefix) {
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/formats/ConsoleLine.java:30]: Do not refactor what is not necessary.

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java [hunk 23]
```diff
@@ -158,7 +157,7 @@ private void checkIdentities() {
         final Collection<? extends IdentitySchemaNode> newIdentities = newModule.getIdentities();
         for (final IdentitySchemaNode oldIdentity : oldIdentities) {
             if (isIdentityNotFound(newIdentities, oldIdentity)) {
-                errors.add(missingIdentityError().updateInformation(DONT_EXISTS, oldIdentity.getPath().toString()));
+                errors.add(missingIdentityError().updateInformation(DONT_EXISTS, oldIdentity.getQName().toString()));
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:160]: Why? Identity statement is a root node which  have only one qName. If I'm not mistaken.
```
  identity test-identity {
    base interface-type;
    description "test-identity.";
  }
```
getPath() -> AbsoluteSchemaPath{path=[(changed:urn:ietf:params:xml:ns:yang:ietf-interfaces?revision=2019-02-20)test-identity]}
getQname() - (changed:urn:ietf:params:xml:ns:yang:ietf-interfaces?revision=2019-02-20)test-identity

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java [hunk 24]
```diff
@@ -245,7 +237,7 @@ private void checkRPCs() {
             }
             if (!rpcFound) {
                 errors.add(missingNodeError().updateInformation("missing rpc node",
-                        getPathFromSchemaIS(oldSchemaIS)));
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:248]: Yes it is. Is there any difference when is used oldSchemaIS.toSchemaNodeIdentifier() instead of Absolute.of(oldRPC.getQname()) ?

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java [hunk 25]
```diff
@@ -383,9 +373,9 @@ private void checkMaxElements(final DataSchemaNode oldNode, final DataSchemaNode
             final Integer newMaxElements = newElementCountConstraint.get().getMaxElements();
             final Integer oldMaxElements = oldElementCountConstraint.get().getMaxElements();
             if (newMaxElements != null && oldMaxElements != null && newMaxElements < oldMaxElements) {
-                errors.add(maxElementsError().updateInformation(getPathFromSchemaIS(this.newSchemaIS)
+                errors.add(maxElementsError().updateInformation(this.newSchemaIS.toSchemaNodeIdentifier()
                                 + MAX_ELEMENTS + newMaxElements,
-                        getPathFromSchemaIS(this.oldSchemaIS) + MAX_ELEMENTS + oldMaxElements));
+                        this.oldSchemaIS.toSchemaNodeIdentifier() + MAX_ELEMENTS + oldMaxElements));
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:378]: SchemaInferenceStack is initialized inside constructor. This private method is called by public method validate().
```java
    public void validate() {
        checkName();
        checkNamespace();
        checkRevision();
        if (is7950) {
            checkIdentities();
        }
        final Collection<? extends DataSchemaNode> childNodes = oldModule.getChildNodes();
        findNodesRecursively(childNodes);

        checkNotifications();
        checkAugmentations();
        checkRPCs();
        checkTypeDefs();
    }
```

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java [hunk 26]
```diff
@@ -591,8 +572,7 @@ private void checkLength(final LengthRestrictedTypeDefinition<?> oldNode,
         if (oldLengths.isPresent()) {
             if (newLengths.isEmpty()) {
                 if (!oldLengths.get().getAllowedRanges().equals(INTEGER_ALLOWED_RANGES)) {
-                    errors.add(lengthError().updateInformation(DONT_EXISTS,
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:594]: Takes up one extra line, but I can revert it or move it to "Refactore codestyle, unify methods format & add final keyword" commit

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java (hunk 6 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java:76]: Yes, but RPCs have input and output which is required to track with schemaIS.

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java (hunk 7 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java:79]: Yes, but Notification can have child nodes, which is required to track with schemaIS.

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java (hunk 9 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java:67]: These methods here used recursive and go through all child nodes in schemaTree.

File: src/main/java/io/lighty/yang/validator/formats/Line.java (hunk 12 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/Line.java:83]: Sorry don't understand, Here is removed workaround with choice nodes. Choice nodes should be removed from path before calling `context.findDataTreeChild(qNames)`. This was resolved with SchemaIS.

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java (hunk 14 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java:299]: It's marked for removal. This will need to be done soon. Should I remove it?

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 27]
```diff
@@ -98,7 +98,7 @@ private void printLines(final List<Line> lines) {
     }
 
     private List<Line> getNotificationsLines(final SingletonListInitializer singletonListInitializer,
-            final Module module, final List<Integer> removeChoiceQnames,
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java]: Parameter removeChoiceQnames was removed. It was not needed anymore when SchemaIS was used.

File: src/main/java/io/lighty/yang/validator/simplify/stream/TrackingXmlParserStream.java (hunk 21 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/simplify/stream/TrackingXmlParserStream.java:138]: Well, here is load whole model from XML with recursive. So for tracking all parts and creating correct SchemaTree it is needed.

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java (hunk 14 above)
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java:299]: No. Yes.

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java (hunk 23 above)
[the remark this answers is not in the document]
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:160]: OK

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java [hunk 28]
```diff
@@ -187,15 +186,14 @@ private void checkBaseIdentities(final IdentitySchemaNode oldIdentity, final Ide
...
-                            .equals(newBaseIdentity.getQName().getLocalName())) {
+                    if (oldBaseIdentity.getQName().getLocalName().equals(newBaseIdentity.getQName().getLocalName())) {
                         identityBaseNotFound = false;
                         break;
                     }
                 }
                 if (identityBaseNotFound) {
                     errors.add(missingBaseIdentityError()
-                            .updateInformation(DONT_EXISTS, oldBaseIdentity.getPath().toString()));
+                            .updateInformation(DONT_EXISTS, oldBaseIdentity.getQName().toString()));
```
[the remark this answers is not in the document]
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:196]: OK

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java (hunk 24 above)
[the remark this answers is not in the document]
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:248]: OK

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java [hunk 29]
```diff
@@ -295,20 +287,18 @@ private void checkNotifications() {
             }
             if (!notificationFound) {
                 errors.add(missingNodeError().updateInformation("missing notification node",
-                        getPathFromSchemaIS(oldSchemaIS)));
```
[the remark this answers is not in the document]
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:298]: OK

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java [hunk 30]
```diff
@@ -295,20 +287,18 @@ private void checkNotifications() {
...
+                        this.oldSchemaIS.toSchemaNodeIdentifier().toString()));
             }
             this.oldSchemaIS.exit();
         }
     }
 
     private void findNodesRecursively(final Collection<? extends DataSchemaNode> childNodes) {
```
[the remark this answers is not in the document]
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:296]: OK

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java (hunk 25 above)
[the remark this answers is not in the document]
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:378]: OK

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java (hunk 26 above)
[the remark this answers is not in the document]
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java:594]: OK

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java (hunk 6 above)
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java:76]: OK

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java (hunk 7 above)
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java:79]: OK

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java (hunk 9 above)
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java:67]: OK

File: src/main/java/io/lighty/yang/validator/formats/Line.java (hunk 12 above)
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/formats/Line.java:83]: revert this back: `if (dataSchemaNode.effectiveConfig().orElse(Boolean.TRUE)) {`

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/StatementPrinter.java (hunk 16 above)
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/formats/yang/printer/StatementPrinter.java:24]: OK

File: src/main/java/io/lighty/yang/validator/formats/JsonTree.java (hunk 18 above)
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/formats/JsonTree.java:135]: Change it to `clear()`.

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 31]
```diff
@@ -64,20 +64,22 @@ public void emitFormat() {
             final List<Integer> removeChoiceQnames = new ArrayList<>();
             SingletonListInitializer singletonListInitializer = new SingletonListInitializer(1);
 
+            SchemaInferenceStack schemaInferenceStack = SchemaInferenceStack.of(this.schemaContext);
```
[the remark this answers is not in the document]
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java]: OK

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java (hunk 27 above)
[the remark this answers is not in the document]
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java]: OK

File: src/main/java/io/lighty/yang/validator/simplify/stream/TrackingXmlParserStream.java (hunk 21 above)
Comment (ihrasko) [reply src/main/java/io/lighty/yang/validator/simplify/stream/TrackingXmlParserStream.java:138]: OK

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/Util.java (hunk 17 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/yang/printer/Util.java:25]: Unused protected method. Done in cleanup.

File: src/main/java/io/lighty/yang/validator/formats/JsTree.java [hunk 32]
```diff
@@ -162,46 +175,55 @@ private void printLines(final List<Line> lines) {
...
             HtmlLine line = new HtmlLine(new ArrayList<>(ids), lyvNodeData, inputOutputOther, removeChoiceQnames,
                     namespacePrefix);
             lines.add(line);
             resolveChildNodes(lines, new ArrayList<>(ids), node, RpcInputOutput.OTHER, removeChoiceQnames,
-                    Collections.emptyList());
+                    Collections.emptyList(), schemaInferenceStack);
             ids.remove(ids.size() - 1);
+            schemaInferenceStack.exit();
         }
+        schemaInferenceStack.exit();
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsTree.java]: Done

File: src/main/java/io/lighty/yang/validator/formats/JsonTree.java (hunk 20 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsonTree.java]: Done

File: src/main/java/io/lighty/yang/validator/formats/JsonTree.java (hunk 19 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsonTree.java:258]: Done

File: src/main/java/io/lighty/yang/validator/formats/JsonTree.java (hunk 18 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/JsonTree.java]: Done

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java (hunk 14 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java]: Done

File: src/main/java/io/lighty/yang/validator/formats/Line.java (hunk 12 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/Line.java:83]: Done

File: src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java (hunk 5 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/checkupdatefrom/CheckUpdateFrom.java]: Fixed, thanks
